security/secrets_adapter.py

```
@@ -0,0 +1,494 @@
+    if adapter_type is None:
+        adapter_type = os.environ.get("SECRETS_ADAPTER", "dev").lower()
+    
+    adapter_class = _ADAPTERS.get(adapter_type)
+    if adapter_class is None:
+        raise ValueError(f"Unknown adapter type: {adapter_type}. Available: {list(_ADAPTERS.keys())}")
+    
+    # Create new adapter if type specified or none exists
+    if kwargs or _global_adapter is None or not isinstance(_global_adapter, adapter_class):
+        try:
+            _global_adapter = adapter_class(**kwargs)
+        except Exception as e:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)